core/jobs/batch_jobs/skill_validation_jobs.py

```diff
@@ -0,0 +1,94 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for skill models"""
+
+from __future__ import annotations
+
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+(skill_models, ) = models.Registry.import_models([models.NAMES.skill])
+
+
+class GetInvalidSkillMediumRubricsJob(base_jobs.JobBase):
+    """Job that returns skills which have no medium rubric explanation."""
+
+    def filter_invalid_medium_rubrics_explanation(self, skill):
+        """Returns true if the medium rubrics explanation is not present.
+
+        Args:
+            skill: SkillModel. Skill model.
+
+        Returns:
+            bool. True if the medium rubrics explanation is not present.
+        """
+        rubrics = skill[1]
+        for rubric in rubrics:
+            if rubric['difficulty'] == 'Medium':
+                medium_rubrics_explanation = rubric['explanations']
+                return len(medium_rubrics_explanation) == 0
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        total_skills = (
+            self.pipeline
+            | 'Get all SkillModels' >> ndb_io.GetModels(
+                skill_models.SkillModel.get_all(include_deleted=False))
+            | 'Combine skill ids and rubrics' >> beam.Map(
+                lambda skill_model: (
+                    skill_model.id, skill_model.rubrics))
+        )
+
+        skills_having_invalid_medium_rubrics_explanation = (
+            total_skills
+            | 'Filter skills with invalid medium rubrics explanation' >> (
+                beam.Filter(
+                    self.filter_invalid_medium_rubrics_explanation))
```

Comment:
Hi @lkbhitesh07,

I think here you need to pass value, like `self.filter_invalid_medium_rubrics_explanation(...)`.
Here you are only checking `self.filter_invalid_medium_rubrics_explanation` which is always true.
Thanks, I hope it helps!!

Reply:
Thanks for the input @sahiljoster32, I really appreciate this.
Actually, you can say that it's a extra functionality that beam job provides, you don't have to pass any value in parenthesis as it does automatically for you, it iterates over every value in PValue and calls that function and perform automatically.
Thanks

Reply:
Thanks for the info. I'll keep that in mind.